content/learning-paths/servers-and-cloud-computing/cca-container/hello-realm.md

```diff
@@ -49,7 +49,7 @@ int main(int argc, char *argv[])
 Now compile the application as the init executable and package it into the `initramfs`:
```

Comment:
This comment is really about 'hello.c' and is related to the comment at the bottom of this file. 'init' is a special program / process, and exiting it this way is not going to give the reader a nice user experience.

I would suggest to change hello.c to:

```console
#include <fcntl.h>
#include <stdio.h>
#include <unistd.h>

#include <linux/reboot.h>
#include <sys/reboot.h>

int main(int argc, char *argv[])
{
  printf("\n\n");
  printf("******* Hello from the Realm ! *******\n");
  printf("\n\n");

  sleep(30);

  printf("Shutting down the realm (this may take a while)...\n");
  reboot(LINUX_REBOOT_CMD_POWER_OFF);

  while(1);
}
```

Reply:
Agreed, thanks for the nice enhancement with the graceful exit.